networkx/generators/geometric.py

```diff
@@ -844,3 +845,206 @@ def thresholded_random_geometric_graph(
+    clustering, and self-similarity.
+
+    In the one-dimensional $\mathbb{S}^1$ model, a node $i$ is assigned two hidden
+    variables: a hidden degree $\kappa_i$, quantifying its popularity, influance,
+    or importance, and an angular position $\theta_i$ in a one-dimensional
+    sphere (or circle) abstracting the similarity space, where angular distances
+    between nodes are a proxy for their similarity. The radius of the circle is
+    adjusted to $R = N/2\pi$,  where $N$ is the number of nodes, so that the density
+    is set to 1 without loss of generality.
+
+    The connection probability between any pair of nodes takes the form
+    of a gravity law, whose magnitude increases with the product of the
+    hidden degrees (i.e., their combined popularities), and decreases with
+    the angular distance between the two nodes.
+    Specifically, nodes $i$ and $j$ are connected with probability
+
+    $p_{ij} = \frac{1}{1 + \frac{d_{ij}^\beta}{\left(\mu \kappa_i \kappa_j\right)^{\max(1, \beta)}}}$
+
+    where $d_{ij} = R\Delta\theta_{ij}$ is the arc length of the circle between
+    nodes $i$ and $j$ separated by an angular distance $\Delta\theta_{ij}$.
+    Parameters $\mu$ and $\beta$ (also called inverse temperature) control the
+    average degree and the clustering coefficient, respectively.
+
+    It can be shown [2]_ that the model undergoes a structural phase transition
+    at $\beta=1$ so that for $\beta<1$ networks are unclustered in the thermodynamic
+    limit (when $N\to \infty$) whereas for $\beta>1$ the ensemble generates
+    networks with finite clustering coefficient.
+
+    The $\mathbb{S}^1$ model can be expressed as a purely geometric model
+    $\mathbb{H}^2$ in the hyperbolic plane [3]_ by mapping the hidden degree of
+    each node into a radial coordinate as
+
+    $r_i = \hat{R} - \frac{2 \max(1, \beta)}{\beta \zeta} \ln \left(\frac{\kappa_i}{\kappa_0}\right)$
+
+    where $\hat{R}$ is the radius of the hyperbolic disk and $\zeta$ is the curvature,
+
+    $\hat{R} = \frac{2}{\zeta} \ln \left(\frac{N}{\pi}\right)
+    - \frac{2\max(1, \beta)}{\beta \zeta} \ln (\mu \kappa_0^2)$
+
+    The connection probability then reads
+
+    $p_{ij} = \frac{1}{1 + \exp\left({\frac{\beta\zeta}{2} (x_{ij} - \hat{R})}\right)}$
+
+    where
+
+    $x_{ij} = r_i + r_j + \frac{2}{\zeta} \ln \frac{\Delta\theta_{ij}}{2}$
+
+    is a good approximation of the hyperbolic distance between two nodes separated
+    by an angular distance $\Delta\theta_{ij}$ with radial coordinates $r_i$ and $r_j$.
+    For $\beta > 1$, the curvature $\zeta = 1$, for $\beta < 1$, $\zeta = \beta^{-1}$.
+
+
+    Parameters
+    ----------
+    Either n, gamma, mean_degree are provided or kappas. The values of
+    n, gamma, mean_degree (if provided) are used to construct random kappas
+    sampled from the power-law distribution.
+
+    beta: float
+        Inverse temperature, controlling the clustering coefficient.
+    n: int, optional
```

Comment:
For the optional arguments, add `(default: <default value>)` to the type description in the docs.  No word "optional" is needed if there is a default.